sockeye/inference.py

```diff
@@ -132,17 +132,28 @@ class TranslatorInput:
     factors: Optional[List[Tokens]] = None
     source_prefix_tokens: Optional[Tokens] = None
     source_prefix_factors: Optional[List[Tokens]] = None
+    target_prefix_tokens: Optional[Tokens] = None
+    target_prefix_factors: Optional[List[Tokens]] = None
+    use_target_prefix_all_chunks: Optional[bool] = True
+    keep_target_prefix_key: Optional[bool] = True
     restrict_lexicon: Optional[lexicon.TopKLexicon] = None
     constraints: Optional[List[Tokens]] = None
     avoid_list: Optional[List[Tokens]] = None
     pass_through_dict: Optional[Dict] = None
 
     def __str__(self):
-        return 'TranslatorInput(%s, %s, factors=%s, source_prefix_tokens=%s, source_prefix_factors=%s, constraints=%s, avoid=%s)' \
-            % (self.sentence_id, self.tokens, self.factors, self.source_prefix_tokens, self.source_prefix_factors, self.constraints, self.avoid_list)
+        return f'TranslatorInput({self.sentence_id}, {self.tokens}, factors={self.factors}, \
+                                  source_prefix_tokens={self.source_prefix_tokens}, \
```

Comment:
How does this actually look like? I'm asking as it seems like the newline would be included in the string, i.e. are the outputs like this (i.e. lines with a lot of leading whitespace)?
```
TranslatorInput({self.sentence_id}, {self.tokens}, factors={self.factors},
                                  source_prefix_tokens={self.source_prefix_tokens},
```
?

For a long multi-line string one could do:
```
        return (f'TranslatorInput({self.sentence_id}, {self.tokens}, factors={self.factors}, 
                                  f'source_prefix_tokens={self.source_prefix_tokens}', 
                                  f'source_prefix_factors={self.source_prefix_factors}', 
                                  f'target_prefix_tokens={self.target_prefix_tokens}', 
                                  f'target_prefix_factors={self.target_prefix_factors}', 
                                  f'use_target_prefix_all_chunks={self.use_target_prefix_all_chunks}', 
                                  f'keep_target_prefix_key={self.keep_target_prefix_key}', 
                                  f'constraints={self.constraints}', 
                                  f'avoid={self.avoid_list})')
```

Or if the intention is a multi-line string I'd reduce the whitespace indentation.

Reply:
Thanks,
You are right it is with lots of leading spaces. I tested and got something like this: 
`TranslatorInput(36, ['9', '0', '0', '2', '6', '6', '6', '1'], factors=[['h', 'l', 'l', 'l', 'h', 'h', 'h', 'l'], ['h', 'l', 'l', 'l', 'h', 'h', 'h', 'l'], ['h', 'l', 'l', 'l', 'h', 'h', 'h', 'l']],                                   source_prefix_tokens=None,                                   source_prefix_factors=None,                                   target_prefix_tokens=['0', '0', '1', '2', '6', '6'],                                   target_prefix_factors=[[]],                                   use_target_prefix_all_chunks=True,                                   keep_target_prefix_key=True,                                   constraints=None,                                   avoid=None)
`
I changed the code as you suggested. Now it gives good output (no unnecessary extra space). Thanks!
`TranslatorInput(36, ['9', '0', '0', '2', '6', '6', '6', '1'], factors=[['h', 'l', 'l', 'l', 'h', 'h', 'h', 'l'], ['h', 'l', 'l', 'l', 'h', 'h', 'h', 'l'], ['h', 'l', 'l', 'l', 'h', 'h', 'h', 'l']]", 'source_prefix_tokens=None', 'source_prefix_factors=None', "target_prefix_tokens=['0', '0', '1', '2', '6', '6']", 'target_prefix_factors=[[]]', 'use_target_prefix_all_chunks=True', 'keep_target_prefix_key=True', 'constraints=None', 'avoid=None)')
`

Reply:
OK Pylint does not like having multiple lines of texts.

To make it simple let us put everything in one single line instead. This is not perfect (very long line) but no need of extra f and Pylint likes it more (than splitting multiple lines.)

`return f'TranslatorInput({self.sentence_id}, {self.tokens}, factors={self.factors}, source_prefix_tokens={self.source_prefix_tokens}, source_prefix_factors={self.source_prefix_factors}, target_prefix_tokens={self.target_prefix_tokens}, target_prefix_factors={self.target_prefix_factors}, use_target_prefix_all_chunks={self.use_target_prefix_all_chunks}, keep_target_prefix_key={self.keep_target_prefix_key}, constraints={self.constraints}, avoid={self.avoid_list})'`

The output is also good, I confirmed. Something like:
`TranslatorInput(36, ['9', '0', '0', '2', '6', '6', '6', '1'], factors=[['h', 'l', 'l', 'l', 'h', 'h', 'h', 'l'], ['h', 'l', 'l', 'l', 'h', 'h', 'h', 'l'], ['h', 'l', 'l', 'l', 'h', 'h', 'h', 'l']]", 'source_prefix_tokens=None', 'source_prefix_factors=None', "target_prefix_tokens=['0', '0', '1', '2', '6', '6']", 'target_prefix_factors=[[]]', 'use_target_prefix_all_chunks=True', 'keep_target_prefix_key=True', 'constraints=None', 'avoid=None)')`

Reply:
oh ok. Thanks for checking!